torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -1120,8 +1121,11 @@ std::pair<NvrtcFunction, std::string> nvrtcCompile(
     int effective_max_reg_per_warp = max_reg_per_warp /
         reg_allocation_granularity * reg_allocation_granularity;
     // The maximum possible count allowed by ptxas is 255
-    max_register = static_cast<uint32_t>(
-        std::min(effective_max_reg_per_warp / warp_size, 255));
+    max_register = std::min(effective_max_reg_per_warp / warp_size, 255);
+    // heuristic suggested max register count to achieve a good occupancy ratio
+    if (max_register_heuristic > 32) {
```

Comment:
Why `> 32`?

Reply:
I used `max_register_heuristic > 32` as a safe guide. will remove. then we must make sure `max_register_heuristic` is initilized to 255 rather than -1 (uninitilized).
"honor max_register_heuristic no matter if opt_block_size has a value." **done**.